agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -234,245 +134,318 @@ public CallReplacementTransformer() {}
-          MethodDef orig = new MethodDef(cname + "." + mname, args);
-          if (debug_class)
-            System.out.printf("looking for %s in map %s%n", orig, current_method_map);
-          ReplacementClass call = current_method_map.get(orig);
-          if (call != null) {
-            call.cnt++;
-            String classname = call.method_class;
-            String methodname = mname;
-            debug_map.log(
-                "Replacing method %s.%s (%s) with %s.%s%n",
-                cname, mname, ArraysMDE.toString(args), classname, methodname);
-            il.append(
-                ifact.createInvoke(
-                    classname, methodname, iv.getReturnType(pgen), new_args, Const.INVOKESTATIC));
-          }
-          return (il);
-        }
+        /*
+         * These are invocations where an implicit argument occurs in the call.
+         */
+        Type instanceType = invocation.getReferenceType(pgen);
+        @UnknownInterned
+        @UnknownKeyFor
+        @NonNull
+        @Initialized
+        Type[] arguments = BCELUtil.insert_type(instanceType, invocation.getArgumentTypes(pgen));
+        instructionList.append(
+            ifact.createInvoke(
+                call.getClassname(),
+                call.getName(),
+                invocation.getReturnType(pgen),
+                arguments,
+                Const.INVOKESTATIC));
+        break;
+
+      case Const.INVOKESTATIC:
+        instructionList.append(
+            ifact.createInvoke(
+                call.getClassname(),
+                call.getName(),
+                invocation.getReturnType(pgen),
+                invocation.getArgumentTypes(pgen),
+                Const.INVOKESTATIC));
+        break;
 
       default:
-        return (null);
+        return null;
+    }
+    return instructionList;
+  }
+
+  /**
+   * Returns the replacement method for the given method if one is determined by a method, class or
+   * package replacement, or null otherwise.
+   *
+   * <p>Class or package replacements are represented as strings in the {@link
+   * #prefixReplacementMap}. When the argument method belongs to one of these classes/packages, a
+   * new {@link MethodDef} replacement is constructed for the method and is added to the {@link
+   * #replacementMap}.
+   *
+   * @param orig the method to replace, must not be null
+   * @return the replacement method, null if there is none
+   */
+  private MethodDef getReplacement(MethodDef orig) {
+    MethodDef replacement = replacementMap.get(orig);
+    if (replacement != null) {
+      return replacement;
+    }
+    // check for a class or package prefix.
+    String prefix = orig.getClassname();
+    String prefixReplacement = prefixReplacementMap.get(prefix);
+    // if prefixReplacement not null, the class name has a replacement
+    // otherwise, strip off classname and search for package
+    if (prefixReplacement == null) {
+      int dotPos = prefix.lastIndexOf('.');
```

Comment:
This use of `lastIndexOf` (and the use of `get` 3 lines later) indicates that prefixReplacement isn't about prefixes, but about exact matches for complete package names.  The names and documentation of the fields needs to be clarified.